### PR TITLE
7D2D a21 Update

### PR DESCRIPTION
**Pull request by IceOfWraith**
Tentative changes for the upcoming a21 release.
Added the following settings:

- WebDashboardEnabled
- WebDashboardPort
- WebDashboardUrl
- EnableMapRendering
- AllowSpawnNearBackpack
- MaxQueuedMeshLayers
- MaxChunkAge
- SaveDataLimit

Updates the Control Panel Password to be more secure.
Removes the unused "Additional Port".
Cleans up the config file handling to use a variable in the updates and metaconfig.

### CODE REVIEW

File: seven-days-to-die.kvp [hunk 1]
```diff
@@ -37,7 +37,7 @@ App.HasReadableConsole=True
...
 App.ApplicationIPBinding=0.0.0.0
-App.Ports=[{"Protocol":"Both","Port":26900,"Offset":0,"Range":1,"Ref":"ServerPort","Name":"Server and Steam Port","Description":"Main game traffic and Steam Query port","ChildPorts":null},{"Protocol":"TCP","Port":26901,"Offset":0,"Range":1,"Ref":"ControlPanelPort","Name":"Control Panel Port","Description":"Web panel access port","ChildPorts":null},{"Protocol":"UDP","Port":26902,"Offset":0,"Range":1,"Ref":"AdditionalPort","Name":"Server Additional Port","Description":"Unknown use at this time","ChildPorts":null},{"Protocol":"TCP","Port":27020,"Offset":0,"Range":1,"Ref":"RemoteAdminPort","Name":"RCON Port","Description":"RCON traffic port","ChildPorts":null}]
+App.Ports=[{"Protocol":"Both","Port":26900,"Offset":0,"Range":1,"Ref":"ServerPort","Name":"Server and Steam Port","Description":"Main game traffic and Steam Query port","ChildPorts":null},{"Protocol":"TCP","Port":8080,"Offset":0,"Range":1,"Ref":"ControlPanelPort","Name":"Control Panel Port","Description":"Web panel access port","ChildPorts":null},{"Protocol":"TCP","Port":8081,"Offset":0,"Range":1,"Ref":"RemoteAdminPort","Name":"RCON Port","Description":"RCON traffic port","ChildPorts":null}]
```
Comment (Greelan) [inline seven-days-to-die.kvp:40]: I don't think it is a good idea configuring default ports that are in the AMP service port range

File: seven-days-to-dieconfig.json [hunk 2]
```diff
@@ -673,6 +716,19 @@
...
+        "DisplayName":"Max Queued Mesh Layers",
+        "Category":"7d2d - Gameplay Settings",
+        "Description":"Maximum amount of Chunk mesh layers that can be enqueued during mesh generation. Reducing this will improve memory usage but may increase Chunk generation time.",
+        "Keywords":"Max,Queued,Mesh,Layers",
+        "FieldName":"MaxQueuedMeshLayers",
+        "InputType":"number",
+        "IsFlagArgument":false,
+        "ParamFieldName":"/ServerSettings/property[@name='MaxQueuedMeshLayers']/@value",
+        "IncludeInCommandLine":false,
+        "DefaultValue":"1000",
```
Comment (Greelan) [inline seven-days-to-dieconfig.json:729]: Suggest a Placeholder as well

File: seven-days-to-dieconfig.json [hunk 3]
```diff
@@ -1119,6 +1175,34 @@
...
+        "DisplayName":"Max Chunk Age",
+        "Category":"7d2d - Server Settings",
+        "Description":"The number of in-game days which must pass since visiting a chunk before it will reset to its original state if not revisited or protected (e.g. by a land claim or bedroll being in close proximity).",
+        "Keywords":"Max,Chunk,Age",
+        "FieldName":"MaxChunkAge",
+        "InputType":"number",
+        "IsFlagArgument":false,
+        "ParamFieldName":"/ServerSettings/property[@name='MaxChunkAge']/@value",
+        "IncludeInCommandLine":false,
+        "DefaultValue":"-1",
```
Comment (Greelan) [inline seven-days-to-dieconfig.json:1188]: Suggest a Placeholder as well

File: seven-days-to-dieconfig.json [hunk 4]
```diff
@@ -1119,6 +1175,34 @@
...
+        "DisplayName":"Save Data Limit",
+        "Category":"7d2d - Server Settings",
+        "Description":"The maximum disk space allowance for each saved game in megabytes (MB). Saved chunks may be forceably reset to their original states to free up space when this limit is reached. Negative values disable the limit.",
+        "Keywords":"Save,Data,Limit",
+        "FieldName":"SaveDataLimit",
+        "InputType":"number",
+        "IsFlagArgument":false,
+        "ParamFieldName":"/ServerSettings/property[@name='SaveDataLimit']/@value",
+        "IncludeInCommandLine":false,
+        "DefaultValue":"-1",
```
Comment (Greelan) [inline seven-days-to-dieconfig.json:1202]: You guessed it - suggest a Placeholder as well

**Review by Greelan — COMMENTED** [review]
Some minor suggestions based on a desktop review

File: seven-days-to-die.kvp (hunk 1 above)
Comment (IceOfWraith) [reply seven-days-to-die.kvp:40]: It shouldn't have any impact. AMP will assign it like usual only using the application available ports.

Comment (Greelan) [reply seven-days-to-die.kvp]: Acknowledge that. Still not a fan. And I don't rule out AMP doing something funky. Your call.

Comment (IceOfWraith) [reply seven-days-to-die.kvp]: I already made the changes if you want to review again.

**Review by Greelan — DISMISSED** [review]

**Comment by Greelan** [comment]
Only other thought is that it is not explicit for all the new settings that they only apply to v21+

**Comment by IceOfWraith** [comment]
Had to adjust the default config since 21.0 is a new "latest_experimental" branch. I thought it was the new public originally.

**Comment by IceOfWraith** [comment]
> Only other thought is that it is not explicit for all the new settings that they only apply to v21+

Indeed, this was intentional since it's the "new standard". I think I did mark the old settings though. There are also other settings that don't have designations what version they work on. But that could be added.

**Review by Greelan — APPROVED** [review]

**Pull request merged**